File: java/src/main/java/org/example/schema/PrimaryKey.java
package org.example.schema;

import io.milvus.v2.client.ConnectConfig;
import io.milvus.v2.client.MilvusClientV2;
import io.milvus.v2.common.DataType;
import io.milvus.v2.service.collection.request.AddFieldReq;
import io.milvus.v2.service.collection.request.CreateCollectionReq;

public class PrimaryKey {
    private static final MilvusClientV2 client;
    static {
        client = new MilvusClientV2(ConnectConfig.builder()
                .uri("http://localhost:19530")
                .build());
    }

    private static void Int64PrimaryKey() {
        CreateCollectionReq.CollectionSchema schema = client.createSchema();

        schema.addField(AddFieldReq.builder()
                .fieldName("my_id")
                .dataType(DataType.Int64)
                // highlight-start
                .isPrimaryKey(true)
                .autoID(true)
                // highlight-end
                .build());
    }

    private static void VarcharPrimaryKey() {
        CreateCollectionReq.CollectionSchema schema = client.createSchema();

        schema.addField(AddFieldReq.builder()
                .fieldName("my_id")
                .dataType(DataType.VarChar)
                // highlight-start
                .isPrimaryKey(true)
                .autoID(true)
                .maxLength(512)
                // highlight-end
                .build());
    }

    public static void main(String[] args) {
        Int64PrimaryKey();
        VarcharPrimaryKey();
    }
}

File: python/schema/primary_key.py
from pymilvus import MilvusClient, DataType

schema = MilvusClient.create_schema()

schema.add_field(
    field_name="my_id",
    datatype=DataType.INT64,
    # highlight-start
    is_primary=True,
    auto_id=True,
    # highlight-end
)

schema.add_field(
    field_name="my_id",
    datatype=DataType.VARCHAR,
    # highlight-start
    is_primary=True,
    auto_id=True,
    max_length=512,
    # highlight-end
)